File: gdx/src/com/badlogic/gdx/graphics/glutils/ImmediateModeRenderer.java

package com.badlogic.gdx.graphics.glutils;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.math.Matrix4;

public interface ImmediateModeRenderer {
    void begin(Matrix4 projModelView, int primitiveType);

    void flush();

    void color(Color color);

    void color(float r, float g, float b, float a);

    void color(float colorBits);

    void texCoord(float u, float v);

    void normal(float x, float y, float z);

    void vertex(float x, float y, float z);

    void end();

    int getNumVertices();

    int getMaxVertices();

    void dispose();
}

File: gdx/src/com/badlogic/gdx/graphics/glutils/ImmediateModeRenderer.kt
package com.badlogic.gdx.graphics.glutils

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.math.Matrix4

interface ImmediateModeRenderer {
    fun begin(projModelView: Matrix4?, primitiveType: Int)

    fun flush()

    fun color(color: Color?)

    fun color(r: Float, g: Float, b: Float, a: Float)

    fun color(colorBits: Float)

    fun texCoord(u: Float, v: Float)

    fun normal(x: Float, y: Float, z: Float)

    fun vertex(x: Float, y: Float, z: Float)

    fun end()

    fun getNumVertices(): Int

    fun getMaxVertices(): Int

    fun dispose()
}
